frontend/public/components/overview/project-overview.tsx

```diff
@@ -143,80 +150,84 @@ const ProjectOverviewListItem = ({item, metrics, onClick, selectedItem}) => {
   </div>;
 
   return <ListView.Item
-    onClick={() => isSelected ? onClick({}) : onClick(item)}
+    onClick={() => store.dispatch(UIActions.selectOverviewItem(isSelected ? '' : uid))}
     className={className}
     heading={heading}
     additionalInfo={[additionalInfo]}
   />;
 };
 
-ProjectOverviewListItem.displayName = 'ProjectOverviewListItem';
-
-ProjectOverviewListItem.propTypes = {
-  item: PropTypes.shape({
-    controller: PropTypes.object,
-    obj: PropTypes.object.isRequired,
-    readiness: PropTypes.object,
-  }).isRequired
-};
-
-const ProjectOverviewList = ({items, metrics, onClickItem, selectedItem}) => {
+const ProjectOverviewList: React.SFC<ProjectOverviewListProps> = ({items, metrics, selectedUID}) => {
   const listItems = _.map(items, (item) =>
     <ProjectOverviewListItem
-      key={item.obj.metadata.uid}
       item={item}
+      key={item.obj.metadata.uid}
       metrics={metrics}
-      onClick={onClickItem}
-      selectedItem={selectedItem}
+      selectedUID={selectedUID}
     />
   );
   return <ListView className="project-overview__list">
     {listItems}
   </ListView>;
 };
 
-
-ProjectOverviewList.displayName = 'ProjectOverviewList';
-
-ProjectOverviewList.propTypes = {
-  items: PropTypes.array.isRequired
-};
-
-const ProjectOverviewGroup = ({heading, items, metrics, onClickItem, selectedItem}) =>
+const ProjectOverviewGroup: React.SFC<ProjectOverviewGroupProps> = ({heading, items, metrics, selectedUID}) =>
   <div className="project-overview__group">
     {heading && <h2 className="project-overview__group-heading">{heading}</h2>}
-    <ProjectOverviewList items={items} metrics={metrics} onClickItem={onClickItem} selectedItem={selectedItem} />
+    <ProjectOverviewList
+      items={items}
+      metrics={metrics}
+      selectedUID={selectedUID}
+    />
   </div>;
 
-
-ProjectOverviewGroup.displayName = 'ProjectOverviewGroup';
-
-ProjectOverviewGroup.propTypes = {
-  heading: PropTypes.string,
-  items: PropTypes.array.isRequired
+const stateToProps = ({UI}, ownProps): ProjectOverviewProps => {
+  return {
+    ...ownProps,
+    selectedUID: UI.getIn(['overview', 'selectedUID'])
+  };
 };
```

Comment:
I think this can be

```ts
const stateToProps = ({UI}): ProjectOverviewProps => {
   return {
     selectedUID: UI.getIn(['overview', 'selectedUID']),
   };
 };
```